### PR TITLE
[ROCKETMQ-114] Add javadoc to codebase

### PR DESCRIPTION
**Pull request by lizhanhui**
This is an in progress PR.

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/BrokerStartup.java [hunk 1]
```diff
@@ -95,52 +141,65 @@ public static BrokerController createBrokerController(String[] args) {
...
+
+                // Use a conservative value for slave brokers so that consumer groups won't frequently change targeting
+                // brokers when they are consuming messages at the boundary between being probably resident in physical
+                // memory and swapped out.
                 int ratio = messageStoreConfig.getAccessMessageInMemoryMaxRatio() - 10;
                 messageStoreConfig.setAccessMessageInMemoryMaxRatio(ratio);
             }
 
-            if (commandLine.hasOption('p')) {
+            // FIXME: log should not be null
```
Comment (shroman) [inline broker/src/main/java/org/apache/rocketmq/broker/BrokerStartup.java]: Creating a JIRA issue for this and having its number as a comment is a good practice.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java [hunk 2]
```diff
@@ -112,18 +156,21 @@ private RemotingCommand consumerSendMsgBack(final ChannelHandlerContext ctx, fin
...
         }
 
+        // Make sure message store of the broker is writable.
         if (!PermName.isWriteable(this.brokerController.getBrokerConfig().getBrokerPermission())) {
             response.setCode(ResponseCode.NO_PERMISSION);
             response.setRemark("the broker[" + this.brokerController.getBrokerConfig().getBrokerIP1() + "] sending message is forbidden");
             return response;
         }
 
+        // TODO: we should warn client here because OP may carelessly get things wrong here.
```
Comment (shroman) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java]: I think this needs another JIRA issue too.

**Comment by vongosling** [comment]
What's wrong with this PR. if no other polish, suggest close it:-)

**Comment by shroman** [comment]
@vongosling I suggest to create JIRA issues for all TODOs and FIXMEs here to make sure they are fixed someday.
It's not critical though, if you want to merge, please go ahead ;)

**Comment by lizhanhui** [comment]
Let's close this PR for now and merge parts of docs which helps.

**Pull request closed without merging**